src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2090,4 +2087,15 @@ protected function mollieLogo()
             Mollie_WC_Plugin::getPluginPath('assets/images/mollie-logo.svg')
         );
     }
+
+    /**
+     * Singleton of the class that handles icons (API/fallback)
+     * @return Mollie_WC_Helper_PaymentMethodIconUrl|null
+     */
+    protected function iconFactory()
+    {
+        $paymentMethodsList = Mollie_WC_Plugin::paymentMethodsImages();
+        $factory = Mollie_WC_Helper_PaymentMethodIconUrl::getInstance($paymentMethodsList);
```

Comment:
This is not a factory, the factory is the method but the instance it's a paymentMethodIconUrl instance, I would rename it.